Source/Lib/Codec/EbEncHandle.c

```diff
@@ -1644,80 +1514,9 @@ __attribute__((visibility("default")))
 #endif
 EB_API EB_ERRORTYPE EbDeinitEncoder(EB_COMPONENTTYPE *h265EncComponent)
 {
-    EbEncHandle_t *encHandlePtr = (EbEncHandle_t*)h265EncComponent->pComponentPrivate;
-    EB_ERRORTYPE return_error = EB_ErrorNone;
-    EB_S32              ptrIndex     = 0 ;
-    EbMemoryMapEntry*   memoryEntry  = (EbMemoryMapEntry*)EB_NULL;
-
-    if (encHandlePtr){
-        //Jing: Send signal to quit thread
-        EB_SEND_END_OBJ(encHandlePtr->inputBufferProducerFifoPtrArray, EB_ResourceCoordinationProcessInitCount)
```

Comment:
If the kernel is the in the middle of time consuming task, sending these signals can't really stop the thread.

Reply:
Each kernel thread would quit by EBEncHandleStopThreads() triggered from EncHandle destructor.

Reply:
yeah, it will quit after we call the get object functions.
TerminateThread  is a very dangerous function. we should stop using it.

https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-terminatethread#remarks

It's better to split the stop thread function into two parts.
1. send the quit signal to all threads.
2. wait and close the thread handles.

Reply:
Hi @xuguangxin @Austin-Hu,  I added these logic back and revert TerminateThread. 
the problem is that, `objectPtr is forced to NULL`, causing `objectDestroyer` not being called.
```
#define EB_SEND_END_OBJ(fifoPtrArray, count) \
    for (unsigned int i = 0; i < count; i++) { \
        EbObjectWrapper_t *outputWrapperPtr; \
        EbGetEmptyObject(fifoPtrArray[0], &outputWrapperPtr); \
        outputWrapperPtr->objectPtr = NULL; \  //Junadd: set the pointer to NULL
        EbPostFullObject(outputWrapperPtr); \
    }
```

```
void EBObjectWrapperDctor(EB_PTR p)
{
    EbObjectWrapper_t* wrapper = (EbObjectWrapper_t*)p;
    if (wrapper->objectDestroyer) {
        //customized destroyer
        if (wrapper->objectPtr)        //Junadd: check the pointer here
            wrapper->objectDestroyer(wrapper->objectPtr);
    }
```
My idea is to add another member variable to EbObjectWrapper_t, so that main thread can set it to quit each kernel. What do you think? And what happened to OpenVisualCloud/SVT-AV1#380?

Reply:
Hi both, I made a commit to address this comment. Pls review. Thank you.
No speed regression. There are about two-digits commit mem increase of release build on windows with this commit.